### PR TITLE
:sparkles: Add nonbootable iso api

### PR DESCRIPTION
**Pull request by hroyrh**
**What this PR does / why we need it**:
This PR defines the API for attaching and detaching non-bootable iso image. The Controller implementation for the API is in progress in this pr : https://github.com/metal3-io/baremetal-operator/pull/1594

### CODE REVIEW

File: apis/metal3.io/v1alpha1/dataimage_types.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+// Contains the count of errors and the last error message.
+type DataImageError struct {
+	Count   int    `json:"count"`
+	Message string `json:"message"`
+}
+
+// DataImageSpec defines the desired state of DataImage.
+type DataImageSpec struct {
+	// Important: Run "make" to regenerate code after modifying this file
```
Comment (dtantsur) [inline apis/metal3.io/v1alpha1/dataimage_types.go:38]: nit: please remove

File: apis/metal3.io/v1alpha1/dataimage_types.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+	// AttachedImage shows the currently attached image
+	AttachedImage *AttachedImageReference `json:"attachedImage,omitempty"`
+
+	// Error contains errors related to DataImage attachment/detachment
+	Error *DataImageError `json:"error,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+//+kubebuilder:resource:shortName=dimg
```
Comment (dtantsur) [inline apis/metal3.io/v1alpha1/dataimage_types.go:59]: It's going to be a very rarely used resource, let's maybe not have a short name for it?

**Comment by dtantsur** [comment]
The previous PR modifies a few more API-related files, such as config/base/crds/kustomization.yaml, a sample or RBAC roles. Could you add them too?

**Comment by hroyrh** [comment]
@dtantsur I have updated the base kustomization config, and added cainjection, webhook and sample dataimage resource file. I've left the RBAC configs since they depend on the controller and it will be more appropriate to include them in the other pr.

**Pull request merged**